syntax = "proto2";

package Tests;

message Test1 {
  enum Test1Enum {
    ValueA=1;
    ValueB=2;
    ValueC=3;
  }

  // Varint types
  optional int32 Int32 = 1;
  optional sint32 Sint32 = 2;
  optional uint32 Uint32 = 3;

  optional int64 Int64 = 4;
  optional sint64 Sint64 = 5;
  optional uint64 Uint64 = 6;

  optional bool Bool = 7;

  optional Test1Enum Enum = 8 [default=ValueA];

//  required int32 a = 1;
//  required string b = 2;
//  required int64 c = 4;
//  required float d = 5;
//  required double e = 6;
//  required test1enum f = 7 [default=ValueA];
//  optional bool g = 8;
//  repeated fixed32 h = 9;
//  required sint32 i = 10;
//  required sint64 j = 11;
//  required sfixed64 k = 12;
}